### DISCUSSION TITLE
Suppress css not supported in emails when running `email dev`

### DISCUSSION THREAD

**Discussion by apetta** [resend/react-email#896] [Ideas]
### Goals

1. Prevent css that will not render correctly in an email from being applied on the dev preview provided by `email dev` (e.g. flexbox)

### Background

I think this will improve DX significantly, as many common css properties like flexbox don't work correctly in emails, the dev preview gives a false preview of the email. It's not until you actually send it do you realise what it actually looks like.

### Proposal

Not entirely sure. Perhaps there's a list of CSS that's not supported in emails which can be referenced prior to showing the preview.

**Comment by turbek** [comment]
I just came here to propose something like this. Doing this on the type level would be preferable. Give a TS error when using properties that are not wildly supported (e.g. flexbox).

Going through the examples and seeing that flexbox is quite common in them was a surprise. 

This kind of defeats the purpose of this library imho.

**Reply by apetta** [reply]
Hey, thanks for chiming in on this. 

The TS error approach is interesting, perhaps we could make use of [Can I email](https://www.caniemail.com/features/css-display-flex/) somehow

**Reply by turbek** [reply]
np! Interesting, it looks like it is maintained by hand by https://mastodon.social/@HTeuMeuLeu here: https://github.com/hteumeuleu/caniemail/tree/master/_features
